frontend/public/components/app.jsx

```diff
@@ -311,41 +311,63 @@ const PollConsoleUpdates = React.memo(function PollConsoleUpdates() {
 
   const [isToastOpen, setToastOpen] = React.useState(false);
   const [pluginsChanged, setPluginsChanged] = React.useState(false);
+  const [pluginVersionsChanged, setPluginVersionsChanged] = React.useState(false);
   const [consoleChanged, setConsoleChanged] = React.useState(false);
   const [isFetchingPluginEndpoints, setIsFetchingPluginEndpoints] = React.useState(false);
   const [allPluginEndpointsReady, setAllPluginEndpointsReady] = React.useState(false);
 
   const [pluginsData, setPluginsData] = React.useState();
   const [pluginsError, setPluginsError] = React.useState();
+  const [pluginManifestsData, setPluginManifestsData] = React.useState();
+  const [pluginManifestsError, setPluginManifestsError] = React.useState();
   const safeFetch = React.useCallback(useSafeFetch(), []);
-  const tick = React.useCallback(() => {
+  const updatesTick = React.useCallback(() => {
     safeFetch(`${window.SERVER_FLAGS.basePath}api/check-updates`)
       .then((response) => {
         setPluginsData(response);
         setPluginsError(null);
       })
       .catch(setPluginsError);
   }, [safeFetch]);
-  usePoll(tick, URL_POLL_DEFAULT_DELAY);
+  usePoll(updatesTick, URL_POLL_DEFAULT_DELAY);
+  const fetchPluginManifest = (pluginName) =>
+    coFetchJSON(`${window.SERVER_FLAGS.basePath}api/plugins/${pluginName}/plugin-manifest.json`);
+  const manifestsTick = React.useCallback(() => {
+    const pluginManifests = pluginsData?.plugins?.map((pluginName) =>
+      fetchPluginManifest(pluginName),
+    );
+    Promise.all(pluginManifests)
+      .then((response) => {
+        setPluginManifestsData(response);
+        setPluginManifestsError(null);
+      })
+      .catch(setPluginManifestsError);
+  });
+  usePoll(manifestsTick, URL_POLL_DEFAULT_DELAY);
 
   const prevPluginsDataRef = React.useRef();
+  const prevPluginManifestsDataRef = React.useRef();
   React.useEffect(() => {
     prevPluginsDataRef.current = pluginsData;
+    prevPluginManifestsDataRef.current = pluginManifestsData;
   });
   const prevPluginsData = prevPluginsDataRef.current;
-  const stateInitialized = _.isEmpty(pluginsError) && !_.isEmpty(prevPluginsData);
+  const prevPluginManifestsData = prevPluginManifestsDataRef.current;
+  const stateInitialized =
+    _.isEmpty(pluginsError) &&
+    !_.isEmpty(prevPluginsData) &&
+    _.isEmpty(pluginManifestsError) &&
+    !_.isEmpty(prevPluginManifestsData);
```

Comment:
We should make sure we don't break the console commit version check on clusters that have no plugins. (Presumably `stateInitialized` will always be false if there are no plugins since we don't have any manifest data?)